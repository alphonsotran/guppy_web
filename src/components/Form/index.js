import React, { useState } from 'react';
import axios from 'axios';

import ReadOnlyForm from '../ReadOnlyForm';

import {
  Input,
  Container,
  FormWrapper,
  Button,
  InputForm,
  Label,
} from './styles';

const Form = () => {
  const [input, setInput] = useState('');
  const [guppyUrl, setGuppyUrl] = useState('');
  const [responseStatus, setResponseStatus] = useState(false);

  const registerLink = async (e) => {
    e.preventDefault();
    setResponseStatus(false);
    try {
      const {
        data: { guppy_url },
      } = await axios({
        url: `http://localhost:5000/links`,
        method: 'post',
        headers: {
          'Content-Type': 'application/json; charset=utf-8',
          'Access-Control-Allow-Origin': '*',
        },
        data: { original_url: input },
      });

      setGuppyUrl(guppy_url);
      setResponseStatus(true);
    } catch (error) {
      console.log('Please try again.', error);
    }
  };

  return (
    <Container className="">
      <FormWrapper>
        <InputForm onSubmit={registerLink}>
          <Label htmlFor="link">
            Link
            <Input
              id="link"
              value={input}
              onChange={(e) => setInput(e.target.value)}
              placeholder="https://www.google.com"
            />
          </Label>
          <Button>Submit</Button>
        </InputForm>
      </FormWrapper>

      <ReadOnlyForm value={guppyUrl} status={responseStatus} />
    </Container>
  );
};

export default Form;
