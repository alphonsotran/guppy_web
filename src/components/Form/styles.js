import styled, { css } from 'styled-components';

export const FormWrapper = styled.div`
  display: flex;
  flex: 1;
  align-self: center;
`;

export const Container = styled.div`
  display: flex;
  flex: 1;
  font-size: 1rem;
  padding: 1rem;
  flex-direction: column;
`;

export const Input = styled.input`
  border-bottom: 1px solid black;
  border: 0;
  display: block;
  font-family: Brand, sans-serif;
  font-size: 1rem;
  margin-top: 0.5rem;
  margin: 0;
  padding: 0.5rem;
  width: 18rem;

  ::placeholder {
    color: 'gray';
  }

  ::-ms-clear {
    display: none;
  }
`;

export const Button = styled.button`
  background-color: lightgray;
  border-radius: 1rem;
  border: 1px solid black;
  margin-top: 1rem;
  margin-top: 1rem;
  padding: 0.8rem;
`;

export const InputForm = styled.form`
  flex: 1;
`;

export const Label = styled.label`
  font-weight: 700;
`;

export const ErrorText = styled.p`
  color: red;
  font-weight: 500;
`;

export const ErrorAndButtonWrap = styled.div`
  display: flex;
  flex: 1;
  flex-direction: row-reverse;
  justify-content: space-between;
  align-content: center;
`;
