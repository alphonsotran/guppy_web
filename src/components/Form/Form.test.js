import React from 'react';
import { shallow, mount } from 'enzyme';
import Form from './index';

it('expects to render Form component', () => {
  expect(shallow(<Form />)).toMatchSnapshot();
});
