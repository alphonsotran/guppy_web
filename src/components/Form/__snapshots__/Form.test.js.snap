// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render Form component 1`] = `
<styled.div
  className=""
>
  <styled.div>
    <styled.form
      onSubmit={[Function]}
    >
      <styled.label
        htmlFor="link"
      >
        Link
        <styled.input
          id="link"
          onChange={[Function]}
          placeholder="https://www.google.com"
          value=""
        />
      </styled.label>
      <styled.button>
        Submit
      </styled.button>
    </styled.form>
  </styled.div>
  <ReadOnlyForm
    status={false}
    value=""
  />
</styled.div>
`;
